rundeckapp/grails-app/domain/rundeck/ScheduledExecution.groovy

```diff
@@ -373,31 +374,25 @@ class ScheduledExecution extends ExecutionContext implements EmbeddedJsonData {
             map.notification=[:]
             notifications.each{
                 if(!map.notification[it.eventTrigger]){
-                    map.notification[it.eventTrigger]=[:]
+                    map.notification[it.eventTrigger]=[]
                 }
                 def trigger= map.notification[it.eventTrigger]
                 def map1 = it.toMap()
                 if(map1.type){
-                    //plugin notification with a type
-                    if(!trigger['plugin']){
-                        trigger['plugin']=map1
-                    }else if(trigger['plugin'] instanceof Map){
-                        trigger['plugin']=[trigger.remove('plugin'),map1]
-                    }else if(trigger['plugin'] instanceof Collection){
-                        trigger['plugin'] << map1
+                    Map pluginMap = trigger.find({ notif -> notif.plugin })
```

Comment:
this looks like is not being used

Reply:
It is used when having one or more plugin notifications